OrbitLinuxTracing/ContextSwitchManagerTest.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include "ContextSwitchManager.h"
+
+namespace LinuxTracing {
+
+TEST(ContextSwitchManager, OneCoreMatch) {
+  std::optional<SchedulingSlice> processed_scheduling_slice;
+  ContextSwitchManager context_switch_manager;
+
+  context_switch_manager.ProcessContextSwitchIn(42, 43, 1, 100);
+
+  processed_scheduling_slice =
+      context_switch_manager.ProcessContextSwitchOut(42, 43, 1, 101);
+  ASSERT_TRUE(processed_scheduling_slice.has_value());
+  EXPECT_EQ(processed_scheduling_slice.value().GetPid(), 42);
+  EXPECT_EQ(processed_scheduling_slice.value().GetTid(), 43);
+  EXPECT_EQ(processed_scheduling_slice.value().GetCore(), 1);
+  EXPECT_EQ(processed_scheduling_slice.value().GetInTimestampNs(), 100);
+  EXPECT_EQ(processed_scheduling_slice.value().GetOutTimestampNs(), 101);
+
+  processed_scheduling_slice =
+      context_switch_manager.ProcessContextSwitchOut(42, 43, 1, 102);
+  ASSERT_FALSE(processed_scheduling_slice.has_value());
+}
+
+TEST(ContextSwitchManager, OneCoreThreadExit) {
+  std::optional<SchedulingSlice> processed_scheduling_slice;
+  ContextSwitchManager context_switch_manager;
+
+  context_switch_manager.ProcessContextSwitchIn(42, 43, 1, 100);
```

Comment:
Introduce kPid, kTid, kCore and use them in checks below.

Reply:
This works here because it's a simple test with only one pid, tid, core, but what about when there are more? Is it worth to have kPid1, kPid2, kTid1, ...?

Reply:
just for readabilty - it is hard to understand which is which by only looking at numbers.

Reply:
Done, please have a look.